### ISSUE TITLE
the last ouput of Bi-RNN in TextRNN

### ISSUE THREAD

**Issue by longbowking** [brightmart/text_classification#27]
https://github.com/brightmart/text_classification/blob/68e2fcf57a8dcec7e7d12f78953ed570451f0076/a03_TextRNN/p8_TextRNN_model.py#L67

In the implementation, the final outputs of  Bi-RNN are calculated as the reduce mean among all time stamps. Compared with `output_rnn_last=output_rnn[:,-1,:]`, what is the difference between these two strategies on the impact of the final classification results?

**Comment by brightmart** [comment]
hi,
you post a good question. the difference is whether you use mean average of all timestamp or use last timestamp to make a prediction; the other way is use reduce_max to make a prediction.

so you can choose the way that the best performance among these three methods.

**Comment by longbowking** [comment]
https://github.com/brightmart/text_classification/blob/68e2fcf57a8dcec7e7d12f78953ed570451f0076/a03_TextRNN/p8_TextRNN_model.py#L74-L83
- the final loss is defined as the sum of cross_entropy_loss and L2-loss to penalty large variables, where a hyperparameter `l2_lambda` is used to balance these two items. In the algorithm, as the hyperparameter `l2_lambda` is directly set to `0.0001`, would it be too small(or too large) under some circumstances so that one in the two items loses its contribution to the whole loss? 
- in general, is there any pratically methods to guide me to set values of hyperparameters, e.g. `lambda2`
